src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/VBDefineConstantsEncoding.cs

```diff
@@ -71,28 +83,55 @@ static IEnumerable<string> ReadEntries(string rawText)
                         escaped = false;
                     }
                 }
-                throw new FormatException("Expected valid name value pair.");
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
             }
 
-            static string DecodeCharacters(string value)
+            static string SplitSingleEntry(string entry)
             {
-
-                return value.Replace("/\"", "\"").Replace("/=", "=").Replace("/,", ",").Replace("//", "/");
+                bool escaped = false;
+                for (int i = 0; i < entry.Length; i++)
+                {
+                    if (entry[i] == '=' && !escaped)
+                    {
+                        throw new FormatException($"Expected valid name value pair for defining custom constants.");
+                    }
+                    else if (i == entry.Length - 1)
+                    {
+                        return entry;
+                    }
+                    else if (entry[i] == '/')
+                    {
+                        escaped = !escaped;
+                    }
+                    else
+                    {
+                        escaped = false;
+                    }
+                }
+                throw new FormatException("Expected valid name value pair for defining custom constants.");
```

Comment:
```suggestion
                bool escaped = false;
                foreach (char element in entry.AsEnumerable())
                {
                    if (element == '=' && !escaped)
                    {
                        throw new FormatException($"Expected valid name value pair for defining custom constants.");
                    }

                    escaped = element == '/' && !escaped;
                }
                return entry;
```

I actually don't know how the outer `throw` would even execute anyway. It looks like it was just added to make the compiler happy.